### PR TITLE
feat: Add examples to --help output for all "argocd proj" cmds

### PR DESCRIPTION
**Pull request by mcaci**
Hi again, with this PR I'm trying to address a bigger task to help with the examples from the argocd proj commands for the issue #15690. I understand that the since the task is a bit bigger let me know if there is some examples that I wrote where you see some improvements that I can make. I'll be happy to take your feedback to improve them.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

**Comment by todaywasawesome** [comment]
Great PR! I think we need some additional 👀 because of it's size.

**Comment by mcaci** [comment]
> Great PR! I think we need some additional 👀 because of it's size.

Sure, I'll be waiting feedbacks for it :+1: there is a point I'm going to show you as it confused me and may need an additional review

File: cmd/argocd/commands/project.go [hunk 1]
```diff
@@ -363,8 +417,15 @@ func NewProjectRemoveOrphanedIgnoreCommand(clientOpts *argocdclient.ClientOption
 		name string
 	)
 	var command = &cobra.Command{
-		Use:   "remove-orphaned-ignore PROJECT GROUP KIND NAME",
+		Use:   "remove-orphaned-ignore PROJECT GROUP KIND",
```
Comment (mcaci) [inline cmd/argocd/commands/project.go:420]: This is the place where I would like to be reassured that I didn't make a bad decision: in fact I understood that you can use `NAME` only in combination with the `--name` flag as in `argocd proj add-oprhaned-ignore`, so I removed it from usage and put the example with `--name NAME`. Is my understanding correct?

Comment (crenshaw-dev) [reply cmd/argocd/commands/project.go:420]: Yep, based on this I agree

```go
			if len(args) != 3 {
				c.HelpFunc()(c, args)
				os.Exit(1)
			}
```

Great catch!

File: cmd/argocd/commands/project.go [hunk 2]
```diff
@@ -189,6 +221,10 @@ func NewProjectRemoveSignatureKeyCommand(clientOpts *argocdclient.ClientOptions)
 	var command = &cobra.Command{
 		Use:   "remove-signature-key PROJECT KEY-ID",
 		Short: "Remove GnuPG signature key from project",
+		Example: templates.Examples(`
+			# Remove GnuPG signature key KEY-ID from project PROJECT
+			argocd proj add-signature-key PROJECT KEY-ID
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/project.go:226]: Needs to be switched to `remove-signature-key`.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Just a couple nitpicks

**Review by crenshaw-dev — APPROVED** [review]
Thanks, @mcaci!

**Comment by mcaci** [comment]
No problem, well spotted! :)

**Pull request merged**